### PR TITLE
Name inference for unbind

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #25582 Name inference for min(Tensor, dim?) / max(Tensor, dim?)
* **#25585 Name inference for unbind**

Test Plan:
- new tests [namedtensor ci]

Pull Request resolved: https://github.com/pytorch/pytorch/pull/25585

Differential Revision: [D17185070](https://our.internmc.facebook.com/intern/diff/D17185070)

### CODE REVIEW

File: test/test_namedtensor.py [hunk 1]
```diff
@@ -741,26 +741,29 @@ def get_minmax_output(t):
...
+            Case('sum', True, True, True, True, None),
+            Case('prod', True, False, True, True, None),
+            Case('mean', True, True, True, True, None),
+            Case('var', True, True, True, True, None),
+            Case('std', True, True, True, True, None),
+            Case('std_mean', True, True, False, True, None),
+            Case('var_mean', True, True, False, True, None),
+            Case('min', True, False, True, True, get_minmax_output),
+            Case('max', True, False, True, True, get_minmax_output),
+            Case('unbind', False, False, False, False, None),
```
Comment (nairbv) [inline test/test_namedtensor.py]: if all false, what are we testing for `unbind` ?

File: aten/src/ATen/native/TensorShape.cpp [hunk 2]
```diff
@@ -1055,6 +1055,12 @@ std::vector<Tensor> unbind(const Tensor &self, int64_t dim) {
   return tensors;
 }
 
+#ifdef BUILD_NAMEDTENSOR
+std::vector<Tensor> unbind(const Tensor& self, Dimname dim) {
+  return at::unbind(self, dimname_to_position(self, dim));
```
Comment (nairbv) [inline aten/src/ATen/native/TensorShape.cpp]: I'd expect to see some change in the existing unbind method to remove the one name (if named) from each resulting unbound tensor. Is that not the expected behavior?

File: test/test_namedtensor.py (hunk 1 above)
Comment (zou3519) [reply test/test_namedtensor.py]: There's a default set of tests that run if everything is false. That default set is {test_simple_reduce(op_name, device)} right now.

File: aten/src/ATen/native/TensorShape.cpp (hunk 2 above)
Comment (zou3519) [reply aten/src/ATen/native/TensorShape.cpp]: `unbind` calls `select` which does that already.

**Review by nairbv — APPROVED** [review]

**Pull request closed without merging**